### PR TITLE
Log dmesg & kernel logs

### PR DESCRIPTION
**Pull request by realjoshuau**
CANivore disconnects are not immediately noticed by the Phoenix library but since USB device events are logged, we can tell if a CANivore is temporarily disconnected (this happened to 3173). The CANivore device driver also logs to dmesg.

This PR forks the ConsoleSource from AdvantageKit, extends it to allow it to read any arbitrary file on the system, and sets up the rio to log to that source.

### CODE REVIEW

**Review by chaoticthegreat — APPROVED** [review]
nice

File: src/main/java/frc/robot/utils/FileConsoleSource.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+          return;
+        }
+        if (nextChar != -1) {
+          try {
+            buffer.put((char) nextChar);
+          } catch (BufferOverflowException e) {
+          }
+        } else {
+          // Break read loop, send complete lines to queue
+          break;
```
Comment (ThatXliner) [inline src/main/java/frc/robot/utils/FileConsoleSource.java:99]: so.. the outer most while loop will never end except for function return?

File: src/main/java/frc/robot/utils/FileConsoleSource.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+          output = new String(buffer.array(), 0, i);
+          buffer.rewind();
+          buffer.put(buffer.array(), i + 1, buffer.limit() - i - 1);
+          buffer.position(originalPosition - i - 1);
+          break;
+        }
+      }
+      if (output != null) {
+        try {
+          queue.put(output);
```
Comment (ThatXliner) [inline src/main/java/frc/robot/utils/FileConsoleSource.java:117]: doesn't this only queue a single line?

**Review by ThatXliner — COMMENTED** [review]
didn't test it locally but LGTM

**Pull request merged**